libraries/AP_HAL/utility/RingBuffer.cpp

```diff
@@ -6,9 +6,14 @@
   implement a simple ringbuffer of bytes
  */
 
+static inline uint32_t align_power2(uint32_t u)
+{
+    return 1U << ((sizeof(u) * 8) - __builtin_clz(u - 1));
```

Comment:
This works. Just not sure it's worth here. You can find similar methods to
round to the next power of 2 in kmod and systemd
Em 23 de jun de 2016 1:17 PM, "Murilo Belluzzo" <notifications@github.com>
escreveu:

In libraries/AP_HAL/utility/RingBuffer.cpp
<https://github.com/ArduPilot/ardupilot/pull/4024#discussion_r68265297>:

> @@ -6,9 +6,14 @@
>    implement a simple ringbuffer of bytes
>   */
>
> +static inline uint32_t align_power2(uint32_t u)
> +{
> +    return 1U << ((sizeof(u) * 8) - __builtin_clz(u - 1));

I didn't get why you have to use u - 1. For sizes that are + 1 (like 3, 9,
17) the allocated size is 1 byte less of the requested one. Is this
expected for any reason?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/ArduPilot/ardupilot/pull/4024/files/a48ef48cfde57c63353cfc26a5d5edb099976674..e2df334bad6fc22dcb77846719f7b8c2500f42c0#r68265297>,
or mute the thread
<https://github.com/notifications/unsubscribe/AAB8eKJqL9PD9igohSQ_iKSGbkV22PO5ks5qOrGBgaJpZM4IVtk1>
.